### DISCUSSION TITLE
lockfile is modified when system cmake is available

### DISCUSSION THREAD

**Discussion by qudix** [xmake-io/xmake#3974] [Q&A]
### Xmake Version

v2.7.9+HEAD.c87922676

### Operating System Version and Architecture

Windows 11

### Describe Bug

When building a project requires a package that needs cmake, xmake detects the local cmake binary included with Visual Studio and automatically modifies the `.lock` file to point to a non-existent package version `3.26.0-msvc3`

This is before:
![image](https://github.com/xmake-io/xmake/assets/17361645/d7dbf8bc-6994-489f-8fda-327835621248)

This is after running `xmake`:
![image](https://github.com/xmake-io/xmake/assets/17361645/8a2aec06-becf-41fd-9be8-e16967717a07)

### Expected Behavior

Do not automatically modify the `.lock` file when falling back to system installed tools.

### Project Configuration

[test.zip](https://github.com/xmake-io/xmake/files/11657533/test.zip)

### Additional Information and Error Logs

[test.txt](https://github.com/xmake-io/xmake/files/11657532/test.txt)

**Comment by waruqi** [comment]
If it uses system libraries and tools, then this lock file must be updated, as the system cmake may be updated or uninstalled by the user at any time, and if not updated, it will be broken. 

1. if system cmake is uninstalled -> lock file and project will be broken
2. if system cmake is updated -> The wrong cmake version is stored in the lock file, some project configurations that depend on the cmake version may also be broken

So to ensure that the lock file is completely unchanged, we must use all packages from the xmake-repo repository and completely disable the use of libraries and tools from the system.

That is, the lock file cannot lock libraries and tools from the system because their updates are not controlled by xmake，unless we disable all system libraries and tools.
